apps/chat/channels.py

```diff
@@ -484,8 +514,12 @@ def _transcribe_audio(self, audio: BytesIO) -> str:
                 return speech_service.transcribe_audio(audio)
         return "Unable to transcribe audio"
 
-    def _get_bot_response(self, message: str) -> str:
-        return self.bot.process_input(message, attachments=self.message.attachments)
+    def _get_bot_response(self, message: str) -> ChatMessage:
+        response = self.bot.process_input(message, attachments=self.message.attachments)
+        if ai_message_id := self.bot.get_ai_message_id():
+            return ChatMessage.objects.get(id=ai_message_id)
+
+        return ChatMessage(content=response, message_type=ChatMessageType.AI)
```

Comment:
why not have the bot return the actual chat message?

Reply:
This reminds me, I wanted to check when we don't save the AI message, which seems to be only in the case when using a router (without pipelines). Considering this, we _can_ just return the `ChatMessage`. This will mean _a lot_ of tests should be updated. I'll rather do this as a separate ticket if you're fine with us doing it this way?

Reply:
See [this PR](https://github.com/dimagi/open-chat-studio/pull/1414) where I'm returning the `ChatMessage` instance from `process_input`. The pipeline now also returns the chat message instead of the pipeline state